src/current/v23.1/sso-sql.md

```diff
@@ -1,158 +1,257 @@
+        },
+        {
+          "alg": "RS256",
+          "n": "8ASqcDDTYmFx53uAkWlmqx8qIx0WQacaswAAb7xGx9XL2T71VPtKKgPqoimXzj4fXT2F3opIgOQgGKxkxw2QrAAoejdIud5URrZLRponmbMQO3erG5sd8PC29rjekX8_hvX_AMT3zU8JLKVLFR0xJmFdmIomnarpEvWKmG0SNvbuVveprsve0n1W35uodcRe417vlTNnH9j8fesQekvFf8tIWbHouXDg3B1km4gqZBbQ_MWvGriGBY5sfr7A2d0iNe89Aje7pz1RFLUFOu-u_NyD6RwL6qo4_yetAYIzm02a2KAAq03YPs2LHMVQmkh1LtyeuA6bvf9146cAFY4BCQ",
+          "use": "sig",
+          "e": "AQAB",
+          "kty": "RSA",
+          "kid": "05150a1320b9395b05716877376928509bab44ac"
+        }
+      ]
+    }';
+    ~~~
+
+1. Configure token generation.
 
 {% include_cached copy-clipboard.html %}
 ~~~shell
-cockroach sql --url "postgresql://{SQL_USERNAME}:{JWT_TOKEN}@{CLUSTER_HOST}:26257?options=--crdb:jwt_auth_enabled=true" --certs-dir={CLUSTER_CERT_DIR}
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.enabled = true;
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.use_token = id_token;
 ~~~
 
-~~~txt
-Welcome to the cockroach SQL interface...
-~~~
+## How CockroachDB determines the SQL username from a JWT
+
+1. `server.oidc_authentication.generate_cluster_sso_token.use_token` determines which token to look at (`id_token` or `access_token`).
+1. `server.jwt_authentication.claim` determines which claim in that token, defaulting to picking the token’s subject, but also potentially using email.
+1. `server.identity_map.configuration` maps that claim (along with the token’s issuer) to a SQL username.
+
+## Identity Map configuration
+
+Th cluster setting `server.identity_map.configuration` holds your cluster's identity map configuration, which tells your cluster how to map external identities to SQL users. This is functionally similar to PostgreSQL's [User Name Map](https://www.postgresql.org/docs/current/auth-username-maps.html).
+
+The format of an identity map configuration is a space-separated triple consisting of:
+
+`<external issuer> <external user ID> <SQL username>`
+
+Examples:
+
+- `https://accounts.google.com /^(.*)@cockroachlabs\.com$ \1`
+
+    Maps every cockroachlabs email to a SQL user with the same username (i.e. the part of the email address to the left of the `@`), for example `docs@cockroachlabs.com` becomes `docs` |
+
+- `https://accounts.google.com 1232316645658094244789 roach`
+
+    Maps a single external identity with the hard-coded ID to the SQL user `roach`   |
+
+- `https://accounts.google.com   /^([9-0]*)$   gcp_\1`
+
+    Maps each GCP-provisioned service account to a SQL user named `gcp_{ GCP user ID }`, e.g., `gcp_1234567` for a service account with ID `1234567`. 
```

Comment:
It might be worth including that you can have multiple matches for the same line and picking between them is done by the SQL username in the connection.

Reply:
is this correct @kpatron-cockroachlabs ?

Reply:
LGTM